src/validation.cpp

```diff
@@ -1416,44 +1420,54 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
+                const auto subpackage_wtxid{subpackage_tx->GetWitnessHash()};
+                if (m_pool.exists(GenTxid::Wtxid(subpackage_wtxid))) {
+                    const auto subpackage_it{subpackage_result.m_tx_results.find(subpackage_wtxid)};
+                    results_final.emplace(subpackage_wtxid, subpackage_it->second);
+                    // Erase any previous invalid results for this transaction. For example, this
+                    // could be a low-feerate tx that has just been bumped.
+                    individual_results_nonfinal.erase(subpackage_wtxid);
+                    packageified.Exclude(subpackage_tx);
+                }
+            }
+            // If m_state is valid, we already processed each tx in the loop above.
+            if (subpackage_result.m_state.IsValid()) continue;
+
+            // Another "final" validation result is if the tx failed for a non-policy reason.
+            const auto single_res_it = subpackage_result.m_tx_results.find(wtxid);
+            if (single_res_it != subpackage_result.m_tx_results.end()) {
+                const auto single_res = single_res_it->second;
+                if (single_res.m_state.GetResult() != TxValidationResult::TX_MEMPOOL_POLICY &&
+                    single_res.m_state.GetResult() != TxValidationResult::TX_MISSING_INPUTS) {
+                    // Package validation policy only differs from individual policy in its evaluation
+                    // of feerate. For example, if a transaction fails here due to violation of a
+                    // consensus rule, the result will not change when it is submitted as part of a
+                    // package. Tell the Packageifier that subsequent transactions depending on this one
+                    // should be skipped.
+                    package_state_quit_early.Invalid(PackageValidationResult::PCKG_TX, "transaction failed");
+                    individual_results_nonfinal.emplace(wtxid, single_res);
+                    packageified.Ban(tx);
+                }
                 individual_results_nonfinal.emplace(wtxid, single_res);
-                txns_package_eval.push_back(tx);
             }
         }
     }
 
-    // Quit early because package validation won't change the result or the entire package has
-    // already been submitted. Since this is an ancestor package, if the child is in, that means all
-    // the other transactions in the package are as well. We check for the child by txid because
-    // same-txid-different-witness is an acceptable case for deduplication in the loop above.
-    if (quit_early || m_pool.exists(GenTxid::Txid(child->GetHash()))) {
+    const auto txns_package_eval{packageified.GetAncestorSet(child)};
+    // If txns_package_eval is std::nullopt, the last tx's result was pre-filled with missing
+    // inputs. If txns_package_eval is empty, all transactions have already passed.
+    if (!txns_package_eval || txns_package_eval->empty()) {
```

Comment:
936c7f170bc68ec583ae026ddbaf197ca6cd0792
Could you help me understand how the latter condition can be possible for this transaction?

Reply:
I assume you're asking "how can `txns_package_eval` be empty?" 
If all of the transactions were already in the mempool, they have each been `Exclude`'d. So `GetAncestorSet(child)` doesn't returns a vector of all the non-excluded ancestors, which is an empty vector.